rllib/tests/test_worker_failures.py

```diff
@@ -381,6 +383,138 @@ def test_workers_fatal_but_recover(self):
                 )
             )
 
+    def test_policies_are_restored_on_recovered_worker(self):
+        class AddPolicyCallback(DefaultCallbacks):
+            def __init__(self):
+                super().__init__()
+
+            def on_algorithm_init(self, *, algorithm, **kwargs):
+                # Add a custom policy to algorithm
+                algorithm.add_policy(
+                    policy_id="test_policy",
+                    policy_cls=PGTorchPolicy,
+                    observation_space=gym.spaces.Box(low=0, high=1, shape=(8,)),
+                    action_space=gym.spaces.Discrete(2),
+                    config={},
+                    policy_state=None,
+                    evaluation_workers=True,
+                )
+
+        # Counter that will survive restarts.
+        COUNTER_NAME = "test_policies_are_restored_on_recovered_worker"
+        counter = Counter.options(name=COUNTER_NAME).remote()
+
+        config = {
```

Comment:
Sorry, but we should try to do everything now using AlgoConfig objects. I know, we still have tons of actual python dicts flying around in our test cases, but let's not add new ones anymore.

Reply:
ok, I went ahead and migrated all the config dicts in this file.
like it actually, since we don't have to pass algorithm name anymore.